### PR TITLE
Add support for POST, GET, PATCH and DELETE /datasets

### PR DESCRIPTION
**Pull request by magnurud**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Changelog 
 
+## Version 1.4.0 - 2021-06-28
+
+- Added CreateDataset
+- Added GetDatasets
```
Comment (augustak) [inline CHANGELOG.md:6]:
```suggestion
- Added ListDatasets
```

File: Lucidtech/Las/Client.cs [hunk 2]
```diff
@@ -294,12 +288,16 @@ public Client(Credentials credentials)
                 {"contentType", contentType},
             };
 
+            if (!string.IsNullOrEmpty(batchId)) {
+                body.Add("batchId", batchId);
+            }
+
             if(consentId != null) {
                 body.Add("consentId", consentId);
             }
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:297]: Should this have the same 'IsNullOrEmpty' check?

File: Lucidtech/Las/Client.cs [hunk 3]
```diff
@@ -439,6 +445,10 @@ public object UpdateDocument(string documentId, List<Dictionary<string, string>>
                 queryParams.Add("consentId", consentId);
             }
 
+            if (datasetId != null) {
+                queryParams.Add("datasetId", datasetId);
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:449]: IsNullOrEmpty?

File: Lucidtech/Las/Client.cs [hunk 4]
```diff
@@ -531,6 +541,116 @@ public object CreateBatch(string? name = null, string? description = null)
...
+        /// <example>
+        /// Create a new dataset with the provided description.
+        /// on the document specified by '&lt;datasetId&gt;'
+        /// <code>
+        /// Client client = new Client();
+        /// var response = client.CreateDataset("Data gathered from the Mars Rover Invoice Scan Mission");
+        /// </code>
+        /// </example>
+        /// <param name="name">Name of the dataset</param>
+        /// <param name="description"> A brief description of the purpose of the dataset </param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:557]:
```suggestion
        /// <param name="description"> A brief description of the dataset </param>
```

File: Lucidtech/Las/Client.cs [hunk 5]
```diff
@@ -531,6 +541,116 @@ public object CreateBatch(string? name = null, string? description = null)
...
+        /// <returns>
+        /// A deserialized object that can be interpreted as a Dictionary with the fields datasetId and description.
+        /// datasetId can be used as an input when posting documents to make them a part of this dataset.
+        /// </returns>
+        public object CreateDataset(string? name = null, string? description = null)
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:562]: Isn't name and description usually provided in an 'attributes' object?

**Review by augustak — CHANGES_REQUESTED** [review]

File: Lucidtech/Las/Client.cs (hunk 5 above)
Comment (magnurud) [reply Lucidtech/Las/Client.cs:562]: Current code is a bit inconsistent, but CreateBatch had name and description provided explicitly, so I would suggest we do the same here for now.

File: Lucidtech/Las/Client.cs (hunk 3 above)
Comment (magnurud) [reply Lucidtech/Las/Client.cs:449]: It could be, currently we're a bit inconsistent here.

**Review by augustak — APPROVED** [review]

**Pull request merged**